tools/eslint-plugin/dist/rules/require-version-in-deprecation.js

```diff
@@ -7,41 +7,18 @@
 
 "use strict";
 
-const { getParserServices } = require("./utils/parser");
-const ts = require("typescript");
-
 module.exports = {
   create(context) {
-
-    const parserServices = getParserServices(context);
-
-    function checkJsDoc(declaration, node) {
-      let tags = ts.getJSDocTags(declaration);
-
-      if (!tags || tags.length == 0) {
-        return;
-      }
-      for (let tag of tags) {
-        if (tag.tagName.escapedText == "deprecated") {
-
-          if (tag.comment == undefined) {
-            context.report(node, "@deprecated version and description missing");
-          }
-          else if (!(/\d+(\.(\d|x))+.{5,}/.test(String(tag.comment)))) {
-            context.report(node, "@deprecated missing version or description.");
+    return {
+      Program(node) {
+        for (let comment of node.comments) {
+          if (/@deprecated/.test(comment.value)) {
```

Comment:
Nit: The levels of nesting could be reduced. You could check the second condition with the first, or perhaps invert the conditions and return early. Another option would be to extract the code in each iteration to a function.

Reply:
Switched to using capture groups two avoid two regexes.